### ISSUE TITLE
Only listen to target element CSS events.

### ISSUE THREAD

**Issue by minht11** [otonashixav/solid-flip#1]
Imagine you have a button which toggles transition on click, something like this:
```jsx
<Transition>
  <div id='animated-container'> // This has animation of 300ms
      <button /> // This button has background transition of 100ms on hover
    </div>
<Transition>
```
Since button transition finishes first these listeners inside [cssExit](https://github.com/otonashixav/solid-flip/blob/4361a608e3342779acc1e8bd75d3991e4c3a7481/src/helpers.ts#L165) will catch it and remove element from DOM despite the fact that main animation is still running.
https://github.com/otonashixav/solid-flip/blob/4361a608e3342779acc1e8bd75d3991e4c3a7481/src/helpers.ts#L199-L200

A solution for that is to check if event target is the container and not one of the children.

**Comment by otonashixav** [comment]
Thanks for reporting. Should be fixed by https://github.com/otonashixav/solid-flip/commit/0b275bf914c1ab7ffa5dd3866f114c5194094e7f; I've published the fixed version.

**Comment by minht11** [comment]
Thanks latest version works as expected.

**Comment by minht11** [comment]
The 0.7.0 version reintroduced this bug.
From [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget):
>The currentTarget read-only property of the Event interface identifies the current target for the event, as the event traverses the DOM. It always refers to the element to which the event handler has been attached, as opposed to Event.target, which identifies the element on which the event occurred and which may be its descendant.

`currentTarget` only returns element which has an event listener attached to it and not an actual element that fired an event. So if child button fires a transition event, `solid-flip` thinks that an actual parent element already finished animating and removes it.

https://github.com/otonashixav/solid-flip/blob/6ed9c1ca8460682ded8aea20f37639631441d17f/src/integrations.ts#L210-L211

**Comment by otonashixav** [comment]
Fixed in 0.7.3.
